src/lib/components/list/QItem.svelte

```diff
@@ -36,43 +35,42 @@
   $: separatorOptions = getContext<QListProps["separatorOptions"] | undefined>("separator");
 
   $: isActionable = clickable || hasLink || tag === "label";
-
   $: isClickable = isActionable && !disable;
-
   $: isActive = isRouteActive($Quaff.router, to);
-
-  $: classes = createClasses(
-    [
-      $hasMultipleLines && "multiline",
-      dense && "dense",
-      (isActive || (hasLink && active)) && activeClass,
-      (isActive || (hasLink && active)) && "active",
-    ],
-    {
-      component: "q-item",
-      quaffClasses: [linkClasses],
-      userClasses,
-    }
-  );
-
-  $: attributes = {
-    class: classes,
-    tabindex: isClickable == true ? Number(tabindex) || 0 : undefined,
-    "aria-disabled": isActionable === true && disable === true ? true : undefined,
-    ...$$restProps,
-  };
 </script>
 
 {#if separatorOptions !== undefined}
   <QSeparator {...separatorOptions} />
 {/if}
 {#if linkAttributes.href !== undefined}
   <!-- svelte-ignore a11y-missing-attribute -->
-  <a use:ripple={{ disable: !isClickable || noRipple }} {...attributes} {...linkAttributes}>
+  <!-- svelte-ignore a11y-no-noninteractive-tabindex -->
+  <a
+    use:ripple={{ disable: !isClickable || noRipple }}
+    class="q-item {linkClasses} {userClasses}"
+    class:q-item--active={isActive || (hasLink && active)}
+    class:q-item--multiline={$hasMultipleLines}
+    class:q-item--dense={dense}
+    class:{activeClass}={isActive || (hasLink && active)}
+    tabindex={isClickable ? Number(tabindex) || 0 : -1}
+    aria-disabled={(isActionable && disable) || undefined}
+    {...linkAttributes}
+    {...$$restProps}
+  >
     <slot />
   </a>
 {:else}
-  <svelte:element this={tag} {...attributes}>
+  <svelte:element
+    this={tag}
+    class="q-item {userClasses}"
+    class:q-item--active={isActive || (hasLink && active)}
+    class:q-item--multiline={$hasMultipleLines}
+    class:q-item--dense={dense}
+    class:{activeClass}={isActive || (hasLink && active)}
+    tabindex={isClickable ? Number(tabindex) || 0 : -1}
```

Comment:
`parseInt(tabindex, 10)` seems more appropriate, unless you want to accept the chance having floats here. But it's up to you, not super important.